frontend/src/modules/EditZing/Components/EmailModal.tsx

```diff
@@ -23,10 +32,42 @@ export const EmailModal = ({
   setEmailSentError,
   handleEmailTimestamp,
 }: EmailModalProps) => {
-  const [selectedTemplate, setSelectedTemplate] = useState<TemplateName>(
-    '' as TemplateName
+  // check if emailing students or groups
+  const recipientType = selectedStudents.length > 0 ? 'student' : 'group'
+
+  // template editor logic
+
+  const [templates, setTemplates] = useState<EmailTemplate[]>([])
+  const [templateName, setTemplateName] = useState('')
+  const [selectedTemplate, setSelectedTemplate] = useState<EmailTemplate>(
```

Comment:
Why are these separate? Doesn't one of these imply the other?

Reply:
That is what I was hoping I think since as you pointed out, selectedTemplate is actually undefined and the radio buttons don't actually work as intended when I try to implement using `selectedTemplate` and `selectedTemplate.name` (for the `currentValue` props for the buttons). I think the main issue is still the fact that selectedTemplate is undefined. I know you said to initialize it at the beginning of the array and how in the old implementation it was initialized to an empty string. 

What would I initialize selectedTemplate to be? I don't think an empty object `{}` works (doesn't type check). And if I don't initialize it, it won't allow me to pass in a possibly undefined object into the other components.

Reply:
> it won't allow me to pass in a possibly undefined object into the other components

What about changing those components to take a possibly undefined object? I think it makes sense for EmailTemplateButtons, as it means "there might be a selected template, or there might not". Then EmailTemplateButtons can determine how to handle that when it shows the radio buttons.

For the email preview, you can use `selectedTemplate!` since by the time the email is being previewed, they must have selected a template, if our app is working correctly. Then you don't need to change how EmailPreview works, it can still take in the email template and not worry about it being undefined.